### PR TITLE
feat(configure): getRenderState for multiple configure widgets

### PR DESCRIPTION
**Pull request by Haroenv**
DX-222

### CODE REVIEW

**Comment by Haroenv** [comment]
All in all, I feel like this explores all the options, meaning that if someone really requires this, they would need to fork connectConfigure, and add a feature very similar to this PR, but that we shouldn't merge it as-is and rather keep the last added configure widget as the only renderState

**Comment by francoischalifour** [comment]
The way that I saw this was that we don't need identifiers. At the end of the day, with the Configure render state, you want to know what are _all_ the search parameters that it applied.

So I think it's fine to merge all the Configure render state properties in there—with a priority to the last props if duplicated, just like how `searchParameters` are sent to Algolia.

Wdyt?

**Comment by Haroenv** [comment]
ah yes good idea, we can merge all searchParameters. Thanks for the idea @francoischalifour

File: src/connectors/configure/connectConfigure.ts [hunk 1]
```diff
@@ -132,13 +132,23 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
...
-          // If we want to merge widgetRenderState of multiple configure widgets,
-          // we should modify this part.
-          configure: this.getWidgetRenderState(renderOptions),
+          configure: {
+            ...widgetRenderState,
+            widgetParams: {
+              ...widgetRenderState.widgetParams,
+              searchParameters: mergeSearchParameters(
+                new algoliasearchHelper.SearchParameters(
+                  renderState.configure?.widgetParams.searchParameters
```
Comment (shortcuts) [inline src/connectors/configure/connectConfigure.ts:144]: Not sure if it's needed but we could add a fallback (`|| {}`) in case configure is `undefined`?

Comment (Haroenv) [reply src/connectors/configure/connectConfigure.ts:144]: it's not needed, since undefined is a valid input for search parameters :)

**Review by eunjae-lee — APPROVED** [review]
It looks good to me.

**Review by shortcuts — APPROVED** [review]

**Pull request merged**